ui/src/cohortReview/cohortReviewList.tsx

```diff
@@ -0,0 +1,220 @@
+import SelectablePaper from "components/selectablePaper";
+import { useTextInputDialog } from "components/textInputDialog";
+import { useSource } from "data/source";
+import { CohortReview } from "data/types";
+import { useCohort, useUnderlay } from "hooks";
+import { useCallback, useEffect } from "react";
+import { Link as RouterLink, useNavigate, useParams } from "react-router-dom";
+import { absoluteCohortURL, cohortReviewURL } from "router";
+import { useImmer } from "use-immer";
+import { useNewReviewDialog } from "./newReviewDialog";
+
+type State = {
+  loading: boolean;
+  reviews?: CohortReview[];
+};
+
+export function CohortReviewList() {
+  const source = useSource();
+  const underlay = useUnderlay();
+  const cohort = useCohort();
+  const navigate = useNavigate();
+  const { reviewId } = useParams<{ reviewId: string }>();
+
+  const [state, updateState] = useImmer<State>({ loading: true });
+
+  const selectedReview = state.reviews?.find(
+    (r, i) => r.id === reviewId || (!reviewId && i === 0)
+  );
+
+  const loadReviews = useCallback(async () => {
+    const reviews = await source.listCohortReviews(cohort.id);
+    updateState((state) => {
+      state.loading = false;
+      state.reviews = reviews;
+    });
+  }, [cohort]);
+
+  useEffect(() => {
+    loadReviews();
+  }, [cohort]);
+
+  const onCreateNewReview = (name: string, size: number) => {
+    updateState((state) => {
+      state.loading = true;
+    });
+    const create = async () => {
+      const review = await source.createCohortReview(name, size, cohort);
+
+      await loadReviews();
+
+      navigate(cohortReviewURL(underlay.name, cohort.id, review.id));
+    };
+    create();
+  };
+
+  const onDeleteReview = () => {
+    if (!selectedReview?.id) {
+      throw new Error("No review selected to delete");
+    }
+
+    updateState((state) => {
+      state.loading = true;
+    });
+    const del = async () => {
+      await source.deleteCohortReview(selectedReview.id, cohort.name);
+      await loadReviews();
+      navigate(cohortReviewURL(underlay.name, cohort.id));
+    };
+    del();
+  };
+
+  const onRenameReview = (name: string) => {
+    updateState((state) => {
+      state.loading = true;
+    });
+    const rename = async () => {
+      if (!selectedReview?.id) {
+        throw new Error("No review selected to rename");
+      }
+
+      await source.renameCohortReview(name, selectedReview.id, cohort.name);
+      await loadReviews();
+    };
+    rename();
+  };
+
+  const [newReviewDialog, showNewReviewDialog] = useNewReviewDialog({
+    onCreate: onCreateNewReview,
+  });
+
+  const [renameReviewDialog, showRenameReviewDialog] = useTextInputDialog({
+    title: `Rename ${selectedReview?.displayName}`,
+    initialText: selectedReview?.displayName,
+    textLabel: "Review name",
+    buttonLabel: "Rename",
+    onConfirm: onRenameReview,
+  });
+
+  return (
+    <>
+      <ActionBar
+        title={`Reviews of ${cohort.name}`}
```

Comment:
Nit: Cohort reviews or Reviews of Cohort {name}?

Reply:
I think that's unnecessarily verbose given the context of how you get to this page and the fact that the names are user selected.